src/sidebar/components/user-menu.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
+
+const bridgeEvents = require('../../shared/bridge-events');
+const { isThirdPartyUser } = require('../util/account-id');
+const serviceConfig = require('../service-config');
+const { withServices } = require('../util/service-context');
+
+const Menu = require('./menu');
+const MenuSection = require('./menu-section');
+const MenuItem = require('./menu-item');
+
+/**
+ * A menu with user and account links.
+ *
+ * This menu will contain different items depending on service configuration,
+ * context and whether the user is first- or third-party.
+ */
+function UserMenu({ auth, bridge, onLogout, serviceUrl, settings }) {
+  const isThirdParty = isThirdPartyUser(auth.userid, settings.authDomain);
+  const service = serviceConfig(settings);
+
+  const serviceSupports = feature => service && !!service[feature];
+
+  const isSelectableProfile =
+    !isThirdParty || serviceSupports('onProfileRequestProvided');
+  const isLogoutEnabled =
+    !isThirdParty || serviceSupports('onLogoutRequestProvided');
+
+  const onProfileSelected = () =>
+    isThirdParty && bridge.call(bridgeEvents.PROFILE_REQUESTED);
+
+  // Generate the correct props for the profile <MenuItem> component
```

Comment:
After refactoring some of the above, the props for the profile MenuItem were getting unwieldy to work with in the JSX so I opted for this approach for populating the props object...is it weird?

Reply:
Using spread syntax is fine for properties which are set conditionally, but I think I would find it easier to read if the static properties were inline. You can mix static and dynamically set properties:

```js
<MenuItem label={auth.displayName} {...menuProps}/>
````

Although this is just JavaScript and you can be as dynamic as you like, I think it tends to be easier to understand what a component does if the returned structure is statically discernable where possible.